lib/spack/spack/cmd/info.py

```diff
@@ -269,6 +271,71 @@ def print_text_info(pkg):
     else:
         color.cprint("    None")
 
+    # Some built-in base packages (e.g., Autotools) define callback (e.g.,
+    # check) inherited by descendant packages. These checks may not result
+    # in build-time testing if the package's build does not implement the
+    # expected functionality (e.g., a 'check' or 'test' targets).
+    #
+    # So the presence of a callback in Spack does not necessarily correspond
+    # to the actual presence of built-time tests for a package.
+    for callbacks, phase in [(pkg.build_time_test_callbacks, 'Build'),
+                             (pkg.install_time_test_callbacks, 'Install')]:
+        color.cprint('')
+        color.cprint(section_title('Available {0} Phase Test Methods:'
+                                   .format(phase)))
+        names = []
+        if callbacks:
+            for name in callbacks:
+                if getattr(pkg, name, False):
+                    names.append(name)
+
+        if names:
+            colify(sorted(names), indent=4)
+        else:
+            color.cprint('    None')
+
+    # PackageBase defines an empty install/smoke test but we want to know
+    # if it has been overridden and, therefore, assumed to be implemented.
```

Comment:
Why the `PackageBase` class defines an empty `test` function? (Apologies if that was discussed in other PRs, feel free to just throw a link to relevant discussion in case)

Reply:
> Why the `PackageBase` class defines an empty `test` function? (Apologies if that was discussed in other PRs, feel free to just throw a link to relevant discussion in case)

IIRC It makes it easier to process stand-alone tests for different types of packages including virtual and compilers but I don't recall if/where we have notes or a discussion on this.  Do you remember @becker33 ?